src/Components/WebAssembly/Server/src/WebAssemblyRazorComponentsBuilderExtensions.cs

```diff
@@ -32,25 +30,21 @@ public static IRazorComponentsBuilder AddWebAssemblyComponents(this IRazorCompon
         return builder;
     }
 
-    private class WebAssemblyEndpointProvider : RenderModeEndpointProvider
+    private class WebAssemblyEndpointProvider(IServiceProvider services) : RenderModeEndpointProvider
     {
-        private readonly IServiceProvider _services;
-        private readonly WebAssemblyComponentsEndpointOptions _options;
-
-        public WebAssemblyEndpointProvider(IServiceProvider services, IOptions<WebAssemblyComponentsEndpointOptions> options)
-        {
-            _services = services;
-            _options = options.Value;
-        }
-
         public override IEnumerable<RouteEndpointBuilder> GetEndpointBuilders(IComponentRenderMode renderMode, IApplicationBuilder applicationBuilder)
         {
             if (renderMode is not WebAssemblyRenderModeWithOptions wasmWithOptions)
             {
+                if (renderMode is WebAssemblyRenderMode)
+                {
+                    throw new InvalidOperationException("Invalid render mode. Use AddWebAssemblyComponents(Action<WebAssemblyComponentsEndpointOptions>) to configure the WebAssembly render mode.");
```

Comment:
Would it make sense to make a similar check in `CircuitEndpointProvider`? We could add an internal type in `AddServerRenderMode` similar to `WebAssemblyRenderModeWithOptions` even if it doesn't configure any options yet.

If we don't do it now, we wouldn't be able to add a similar check in the future without a breaking change if we decide to add a `ServerComponentsEndpointOptions`. It feels weird to allow `ComponentEndpointConventionBuilderHelper.AddRenderMode(builder, new MyServerRenderMode())` to work when `new MyWebAssemblyRenderMode()` cannot.

Reply:
done